File: strings/shortest_distance_to_a_character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        ans = []
        last_pos = -1

        for cur_idx, cur_letter in enumerate(s):
            if cur_letter == c:
                last_pos = cur_idx
                ans.append(0)
            else:
                left = cur_idx - last_pos if last_pos != -1 else float('inf')

                right = float('inf')
                for i in range(cur_idx + 1, len(s)):
                    if s[i] == c:
                        right = i - cur_idx
                        break
                    
                ans.append(min(left, right))

        return ans

File: strings/shortest_distance_to_a_character.rb
def shortest_to_char(s, c)
    ans = []
    hash = { c => -1 }
    str = s.chars

    str.each_with_index do |cur_letter, cur_idx|
        if cur_letter == c
            hash[cur_letter] = cur_idx
            ans << 0
        else
            left = hash[c] == -1 ? str.index(c) : cur_idx - hash[c]
            right = str[cur_idx..].index(c) || left
            ans << [left, right].min
        end
    end